mslib/msui/wms_control.py

```diff
@@ -1598,48 +1483,54 @@ class HSecWMSControlWidget(WMSControlWidget):
-        if self.cbLevelOn.isChecked():
-            s = self.cbLevel.currentText()
-            if s == "":
-                return
-            if self.btGetMap.isEnabled() and self.cbAutoUpdate.isChecked() and not self.layerChangeInProgress:
-                self.btGetMap.click()
-            else:
-                self.view.waypoints_interactor.update()
+        super().level_changed()
+        self.view.waypoints_interactor.update()
 
-    def get_map(self):
+    def get_map(self, layers=None):
         """Slot that retrieves the map and passes the image
            to the view.
         """
         # Get coordinate reference system and bounding box from the map
         # object in the view.
         crs = self.view.get_crs()
         bbox = self.view.getBBOX()
-        if self.wms.version == "1.3.0" and crs.startswith("EPSG") and int(crs[5:]) in axisorder_yx:
-            bbox = (bbox[1], bbox[0], bbox[3], bbox[2])
 
         # Determine the current size of the vertical section plot on the
         # screen in pixels. The image will be retrieved in this size.
         width, height = self.view.get_plot_size_in_px()
-        # Retrieve the image.
-        self.retrieve_image(crs, bbox, None, width, height)
 
-    def display_retrieved_image(self, img, legend_img, layer, style, init_time, valid_time, level):
+        if not layers:
+            layers = [self.multilayers.get_current_layer()]
+
+        layers.sort(key=lambda x: self.multilayers.get_multilayer_priority(x))
+
+        args = []
+        for i, layer in enumerate(layers):
+            transparent = self.cbTransparent.isChecked() if i == 0 else True
+            bbox_tmp = tuple(bbox)
+            wms = self.multilayers.layers[layer.wms_name]["wms"]
+            if wms.version == "1.3.0" and crs.startswith("EPSG") and int(crs[5:]) in axisorder_yx:
+                bbox_tmp = (bbox[1], bbox[0], bbox[3], bbox[2])
+            args.extend(self.retrieve_image(layer, crs, bbox_tmp, None, width, height, transparent))
+
+        self.fetch.emit(args)
+
+    def display_retrieved_image(self, imgs, legend_imgs, layer, style, init_time, valid_time, level):
+        img = self.squash_multiple_images(imgs)
+
         # Plot the image on the view canvas.
-        if style != "":
-            style_title = self.get_layer_object(layer).styles[style]["title"]
-        else:
-            style_title = None
-        self.view.draw_metadata(title=self.get_layer_object(layer).title,
+        style_title = self.multilayers.get_current_layer().get_style()
+        self.view.draw_metadata(title=self.multilayers.get_current_layer().layerobj.title,
                                 init_time=init_time,
                                 valid_time=valid_time,
                                 level=level,
                                 style=style_title)
         self.view.draw_image(img)
-        self.view.draw_legend(legend_img)
+        self.view.draw_legend(legend_imgs[-1])
```

Comment:
this shows the last legend, we need to know that the range of all legends -> data is the same.

Reply:
All legends are shown at once now. Though too many legends may make it quite hard to read, I suppose.

Reply:
Fine for now. I doubt that we we will have more than one in practise, so this doesn't terribly matter. In a second step one might add a feature to select one of multiple legends with a drop-down field. 
We also might think about actually supporting and using the legend feature in our MSS Server, BTW...

Reply:
Hmm, if all legends have the same scale than this does not matter. We need to have checked this,